### PR TITLE
update test docs

### PR DESCRIPTION
**Pull request by BondarenkoStas**

### CODE REVIEW

File: content/testing-yaml/testing.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ aliases:
 ---
 
 Test scripts are added under `scripts` in the [overall architecture](../getting-started/yaml#template), before the build commands.
+You can display test results visually in the build overview if you use expanded form of the script in codemagic.yaml.
+Just include the `test_report` field with a glob pattern matching the test result file location. Supported test report formats are Junit and Flutter `--machine` report.
```
Comment (priitlatt) [inline content/testing-yaml/testing.md:11]:
```suggestion
Just include the `test_report` field with a glob pattern matching the test result file location. Supported test report formats are [Junit XML](https://llg.cubic.org/docs/junit/) and Flutter `--machine` report.
```

File: content/testing-yaml/testing.md [hunk 2]
```diff
@@ -21,37 +23,66 @@ npm test
...
+    adb wait-for-device
+- name: Test
+  script: |
+    set -e
+    ./gradlew connectedAndroidTest
+    adb logcat -d > emulator.log
 ```
 
-## Native iOS
+**Tip:**: you can save the emulator lor with `adb logcat -d > emulator.log` command.
```
Comment (priitlatt) [inline content/testing-yaml/testing.md:46]:
```suggestion
**Tip:**: you can save the emulator log with `adb logcat -d > emulator.log` command.
```

File: content/testing-yaml/testing.md [hunk 3]
```diff
@@ -21,37 +23,66 @@ npm test
...
+      - name: Firebase connected test
+        script: |
+          ./gradlew assembleDevDebugAndroidTest                                            # assemble a test to run it on firebase
+          gcloud firebase test android run \
+            --app "app/dev/release/app.aab" \                                              # path to your actual application
+            --test "$(find app/build/outputs/apk/androidTest -name "*.apk" | head -1)" \   # path to the assembled test
+            --device "model=Nexus5X,version=23" \                                          # a device available from gcloud firebase test android models list
+            --timeout "45m" \
+            --type "instrumentation" \
+            --quiet
```
Comment (priitlatt) [inline content/testing-yaml/testing.md:73]: It should be possible to copy test results XML from GCloud bucket and have nice overview in this case too.

File: content/testing-yaml/testing.md [hunk 4]
```diff
@@ -21,37 +23,66 @@ npm test
...
-    -sdk iphonesimulator \
-    -destination 'platform=iOS Simulator,name=iPhone 6,OS=8.1' \
-    test | xcpretty |& tee "/tmp/xcodetest.log"
+- name: iOS test
+    script: |
+    xcode-project run-tests \
+        --workspace MyAwesomeApp.xcworkspace \
+        --scheme MyAwesomeApp \
+        --device "iPhone 11"
+    test_report: build/ios/test/*.xml
```
Comment (priitlatt) [inline content/testing-yaml/testing.md:85]: Mabe link to docs of `xcode-project run-tests` for more options?

File: content/testing-yaml/testing.md (hunk 1 above)
Comment (priitlatt) [reply content/testing-yaml/testing.md:11]: Perhaps it is worth noting that Junit XML files need to be with `.xml` or `.junit` suffix and Flutter `--machine` reports are expected to have `.json` suffix?

File: content/testing-yaml/testing.md [hunk 5]
```diff
@@ -21,37 +23,66 @@ npm test
...
-./gradlew connectedAndroidTest
+- name: Launch emulator
+  script: |
+    cd $ANDROID_HOME/tools
+    emulator -avd emulator &
+    adb wait-for-device
+- name: Test
+  script: |
+    set -e
+    ./gradlew connectedAndroidTest
```
Comment (priitlatt) [inline content/testing-yaml/testing.md:42]: Doesn't `./gradlew connectedAndroidTest` produce xml report?

File: content/testing-yaml/testing.md (hunk 1 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md]: fixed in 2e33bafc3592fa7408d75ac9459306eb97d56a8d

File: content/testing-yaml/testing.md (hunk 2 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md]: fixed in 2e33bafc3592fa7408d75ac9459306eb97d56a8d

File: content/testing-yaml/testing.md (hunk 4 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md:86]: fixed in 2e33bafc3592fa7408d75ac9459306eb97d56a8d

File: content/testing-yaml/testing.md (hunk 5 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md:42]: added in 2e33bafc3592fa7408d75ac9459306eb97d56a8d

File: content/testing-yaml/testing.md (hunk 3 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md:74]: decided not to do that

**Review by priitlatt — APPROVED** [review]
LGTM now. Maybe Helina can add something.

File: content/testing-yaml/testing.md [hunk 6]
```diff
@@ -7,6 +7,8 @@ aliases:
 ---
 
 Test scripts are added under `scripts` in the [overall architecture](../getting-started/yaml#template), before the build commands.
+You can display test results visually in the build overview if you use expanded form of the script in codemagic.yaml.
```
Comment (helinanever) [inline content/testing-yaml/testing.md:10]: This should be a new paragraph or even in a notebox.
... if you use an expanded form of the script in `codemagic.yaml`.

File: content/testing-yaml/testing.md [hunk 7]
```diff
@@ -7,6 +7,8 @@ aliases:
...
+Just include the `test_report` field with a glob pattern matching the test result file location. Supported test report formats are [Junit XML](https://llg.cubic.org/docs/junit/) and Flutter `--machine` report in `.json` format.
```
Comment (helinanever) [inline content/testing-yaml/testing.md:11]: Supported test report formats are [Junit XML](https://llg.cubic.org/docs/junit/) and `.JSON` for Flutter's `--machine` report.

File: content/testing-yaml/testing.md [hunk 8]
```diff
@@ -21,39 +23,71 @@ npm test
...
+- name: Test
+  script: |
+    set -e
+    ./gradlew connectedAndroidTest
+    adb logcat -d > emulator.log
+  test_report: app/build/outputs/androidTest-results/connected/*.xml
 ```
 
-## Native iOS
+**Tip:**: you can save the emulator log with `adb logcat -d > emulator.log` command.
```
Comment (helinanever) [inline content/testing-yaml/testing.md:47]: There's double colon. 
Also: ...log with the `adb...` command.

File: content/testing-yaml/testing.md [hunk 9]
```diff
@@ -21,39 +23,71 @@ npm test
...
+
+Instrumentation test with Firbase test lab
```
Comment (helinanever) [inline content/testing-yaml/testing.md:49]: Is Firebase test lab support only for native Android?

Comment (helinanever) [reply content/testing-yaml/testing.md:49]: In any case I would prefer formatting all the test types as headings and unifying how we call them (Unit tests, UI tests on emulator, UI tests in Firebase Test Lab). The explanation of UI and instrumentation tests could come after the heading.

File: content/testing-yaml/testing.md [hunk 10]
```diff
@@ -21,39 +23,71 @@ npm test
...
+- name: iOS test
+    script: |
+    xcode-project run-tests \
+        --workspace MyAwesomeApp.xcworkspace \
+        --scheme MyAwesomeApp \
+        --device "iPhone 11"
+    test_report: build/ios/test/*.xml
 ```
 
+Please check [Codemagic CLI tools documentation](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/xcode-project/run-tests.md#run-tests) to learn about more optional arguments to `xcode-project run-tests`.
```
Comment (helinanever) [inline content/testing-yaml/testing.md:89]: .. to learn more about optional arguments to...

File: content/testing-yaml/testing.md (hunk 9 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md:49]: ios is possible as well, will better delete it and add with both ios and android some other time

File: content/testing-yaml/testing.md (hunk 6 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md]: started a new paragraph and fixed in e0d9c8e6222cf49670671fc10c0b345d6644767c

File: content/testing-yaml/testing.md (hunk 7 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md]: fixed in e0d9c8e6222cf49670671fc10c0b345d6644767c

File: content/testing-yaml/testing.md (hunk 8 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md]: fixed in e0d9c8e6222cf49670671fc10c0b345d6644767c

File: content/testing-yaml/testing.md (hunk 10 above)
Comment (BondarenkoStas) [reply content/testing-yaml/testing.md]: fixed in e0d9c8e6222cf49670671fc10c0b345d6644767c

**Review by helinanever — APPROVED** [review]
LGTM

**Pull request merged**